core/includes/config.inc

```diff
@@ -1252,6 +1252,9 @@ class ConfigFileStorage implements ConfigStorageInterface {
       '%directory' => $config_directory,
     )));
     }
+    else {
+      file_save_htaccess($this->directory);
```

Comment:
Shouldn't this be called right after the directory is created as well? That is, move it outside of the `else {}` statement.

Reply:
My understanding is that `if (!file_prepare_directory(` means the directory wasn't created or isn't writable. Which means the `else {` is when the directory is available and writable. Moving the code out of the `else` risks running it when there's no writable directory, right?

Reply:
> Moving the code out of the else risks running it when there's no writable directory, right?

Not in this case, because if the directory doesn't exist we `throw new ConfigStorageException`, which immediately exits the function, so nothing below it will get called.

Reply:
resolved in new PR https://github.com/backdrop/backdrop/pull/3279